manifests/init.pp

```diff
@@ -176,23 +189,17 @@
 ) inherits foreman_proxy_content::params {
   include certs
   include foreman_proxy
-  include foreman_proxy::plugin::pulp
 
-  $pulp_master = $foreman_proxy::plugin::pulp::enabled
-  $pulp = $foreman_proxy::plugin::pulp::pulpnode_enabled
-  $pulpcore_mirror = $foreman_proxy::plugin::pulp::pulpcore_mirror
-  $pulpcore = $foreman_proxy::plugin::pulp::pulpcore_enabled
-
-  $enable_pulp2_rpm = $enable_yum and !($pulpcore and $proxy_pulp_yum_to_pulpcore)
-  $enable_pulp2_iso = $enable_file and !($pulpcore and $proxy_pulp_isos_to_pulpcore)
-  $enable_pulp2_deb = $enable_deb and !($pulpcore and $proxy_pulp_deb_to_pulpcore)
+  $enable_pulp2_rpm = $enable_yum and ($enable_pulp2 and !$proxy_pulp_yum_to_pulpcore)
+  $enable_pulp2_iso = $enable_file and ($enable_pulp2 and !$proxy_pulp_isos_to_pulpcore)
+  $enable_pulp2_deb = $enable_deb and ($enable_pulp2 and !$proxy_pulp_deb_to_pulpcore)
 
   $foreman_proxy_fqdn = $facts['networking']['fqdn']
   $foreman_url = $foreman_proxy::foreman_base_url
-  $reverse_proxy_real = ($pulp or $pulpcore_mirror) or $reverse_proxy
+  $reverse_proxy_real = ($enable_pulp2_mirror or $enable_pulpcore_mirror) or $reverse_proxy
 
   # TODO: doesn't allow deploying a Pulp non-mirror without Foreman
```

Comment:
I'm still thinking how we can achieve this but it's probably easier once we get rid of Pulp 2.